src/Functions.tsx

```diff
@@ -387,9 +426,11 @@ export const createWaypointsTableBody = (
   keys: string[]
 ): JSX.Element => {
   const [state] = stateManager;
+  const originalRows = state.combinedRows;
+  const filteredRows = originalRows.filter(isCombinedRowVisible);
   return (
     <tbody>
-      {state.combinedRows.map((givenCombinedRow, givenIndex) => (
+      {filteredRows.map((givenCombinedRow, givenIndex) => (
```

Comment:
I'm going to guess this is an artifact from *Clean Code*. This goes too far, and actually makes the code more complex while trying to simplify it. I think something like below is readable enough. Most people are pretty comfortable reading chained `map` / `filter` / `reduce` calls.

```typescript

return (
  <tbody>
    {state.combinedRows
    	.filter(e => e.isMarkerVisible)
    	.map((givenCombinedRow, givenIndex) => (
```